microprofile/opentracing-extension/src/main/java/org/wildfly/extension/microprofile/opentracing/JaegerClientMetricsHandler.java

```diff
@@ -0,0 +1,218 @@
+                break;
+            case SPAN_CONTEXT_DECODING_ERRORS:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.decodingErrors).getValue());
+                break;
+            case REPORTER_SPANS_REPORTED_SUCCESS:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.reporterSuccess).getValue());
+                break;
+            case REPORTER_SPANS_REPORTED_FAILED:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.reporterFailure).getValue());
+                break;
+            case REPORTER_SPANS_DROPPED:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.reporterDropped).getValue());
+                break;
+            case REPORTER_QUEUE_LENGTH:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.reporterQueueLength).getValue());
+                break;
+            case SAMPLER_QUERIES_SUCCESS:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.samplerRetrieved).getValue());
+                break;
+            case SAMPLER_QUERIES_FAILED:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.samplerQueryFailure).getValue());
+                break;
+            case SAMPLER_UPDATES_SUCCESS:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.samplerUpdated).getValue());
+                break;
+            case SAMPLER_UPDATES_FAILED:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.samplerParsingFailure).getValue());
+                break;
+            case BAGGAGE_UPDATES_SUCCESS:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.baggageRestrictionsUpdateSuccess).getValue());
+                break;
+            case BAGGAGE_UPDATES_FAILED:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.baggageRestrictionsUpdateSuccess).getValue());
+                break;
+            case BAGGAGE_TRUNCATIONS:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.baggageTruncate).getValue());
+                break;
+            case BAGGAGE_RESTRICTIONS_UPDATES_SUCCESS:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.baggageUpdateSuccess).getValue());
+                break;
+            case BAGGAGE_RESTRICTIONS_UPDATES_FAILED:
+                result.set(((WildflyJaegerMetricsFactory.Metric) metrics.baggageUpdateFailure).getValue());
+                break;
+            default:
+                throw new IllegalStateException(TracingExtensionLogger.ROOT_LOGGER.unknownMetric(metric));
+        }
+
+        context.getResult().set(result);
+    }
+
+    private static ServiceName resolveDeploymentUnitServiceName(OperationContext context, ModelNode operation) {
+        PathAddress address = PathAddress.pathAddress(operation.require(ModelDescriptionConstants.OP_ADDR));
```

Comment:
This should be

PathAddress address = context.getCurrentAddress();

The result is the same but ^^^ is more efficient and is the preferred idiom because it's simpler.

Reply:
done